design/tracing-design.md

```diff
@@ -0,0 +1,170 @@
+            OverallSampling: &envoy_type.Percent{
+                Value: 100,
+            },
+            MaxPathTagLength: wrapperspb.UInt32(256),
+            CustomTags:       []*envoy_trace_v3.CustomTag{
+                {
+                    Tag: "literal",
+                    Type: &envoy_trace_v3.CustomTag_Literal_{
+                        Literal: &envoy_trace_v3.CustomTag_Literal{
+                            Value: "this is literal",
+                        },
+                    },
+                },
+                {
+                    Tag: "environment",
+                    Type: &envoy_trace_v3.CustomTag_Environment_{
+                        Environment: &envoy_trace_v3.CustomTag_Environment{
+                            Name: "HOST",
+                        },
+                    },
+                },
+                {
+                    Tag: "requestHeaderName",
+                    Type: &envoy_trace_v3.CustomTag_RequestHeader{
+                        RequestHeader: &envoy_trace_v3.CustomTag_Header{
+                        Name:         ":path",
+                        },
+                    },
+                },
+            },
+            Provider: &envoy_config_trace_v3.Tracing_Http{
+                Name: "envoy.tracers.opentelemetry",
+                ConfigType: &envoy_config_trace_v3.Tracing_Http_TypedConfig{
+                    TypedConfig: protobuf.MustMarshalAny(&envoy_config_trace_v3.OpenTelemetryConfig{
+                        GrpcService: &envoy_core_v3.GrpcService{
+                            TargetSpecifier: &envoy_core_v3.GrpcService_EnvoyGrpc_{
+                                EnvoyGrpc: &envoy_core_v3.GrpcService_EnvoyGrpc{
+                                    ClusterName: "extension/projectcontour/otle-collector",
+                                    Authority:   "extension/projectcontour/otle-collector",
+                                },
+                            },
+                            Timeout: durationpb.New(time.Millisecond * 50),
+                        },
+                        ServiceName: "contour-envoy-otel.otel-collector",
+                    }),
+                },
+            },
+        },
+...
+```
+
+## Alternatives Considered
+If there are alternative high level or detailed designs that were not pursued they should be called out here with a brief explanation of why they were not pursued.
+
+## Security Considerations
+If this proposal has an impact to the security of the product, its users, or data stored or transmitted via the product, they must be addressed here.
+
+## Compatibility
+A discussion of any compatibility issues that need to be considered
+
+## Implementation
+A description of the implementation, timelines, and any resources that have agreed to contribute.
+
+## Open Issues
+- Envoy supports custom tracing sampling rate and custom label override hcm configuration in route - should contour be supported in the future?
```

Comment:
this makes sense as a future enhancement/follow on after the initial implementation if it is too big a chunk to do in one go